### DISCUSSION TITLE
trivy multi-replcas functioning

### DISCUSSION THREAD

**Discussion by rammanoj** [aquasecurity/trivy-operator#1425] [Help & Support]
I am running 3 replicas of trivy operator on my system. I've deployed a huge load of containers on the system and trivy started and completed the scans successfully 🎉  

However, when I observe the resource that these three pods consume, one of the three is consuming a lot more resources when compared to the other 2. I am currently running with `scanJobsConcurrentLimit` set to 10. PFA RAM snapshot.
![image](https://github.com/aquasecurity/trivy-operator/assets/21082378/ef325628-08a3-48c1-9b55-c2c3afe07fd8)

My assumption is that the load would be distributed across all three pods in a similar fashion, isn't it the scenario? Can I get some context on how is this working?

**Comment by chen-keinan** [comment]
@rammanoj can you share the consuming operator log

**Reply by rammanoj** [reply]
@chen-keinan Right now the pod ending with `2f` is consuming more memory. PFA log of that pod.
[operator.log](https://github.com/aquasecurity/trivy-operator/files/12306496/operator.log)

![image](https://github.com/aquasecurity/trivy-operator/assets/21082378/dbc05433-7aa1-47a2-a59b-8e096d4bff1b)

**Reply by rammanoj** [reply]
@chen-keinan, just curious to know if you find something regarding the above.

**Reply by chen-keinan** [reply]
@rammanoj is the operator reconcile all resources? have you configure the `leader election` ?

**Reply by rammanoj** [reply]
@chen-keinan I have not configured the leader election explicitly. Doesn't it gets auto-enabled? I could not find any other option on the `values.yaml` file to enable the election explicitly. The only option I could see regarding the election is `leaderElectionId`. Is there any way through which I could enable it?

**Reply by chen-keinan** [reply]
Actually it should happen [automatically](https://github.com/aquasecurity/trivy-operator/blob/main/deploy/helm/templates/deployment.yaml#L119) when replica is `gt` than 1

**Reply by rammanoj** [reply]
cool, I have not tampered with its behavior. So, is it consuming more resources because it's acting as a leader and reconciling the resources?

**Comment by chen-keinan** [comment]
only the leader should perform the reconcile

**Reply by rammanoj** [reply]
So, is this pod consuming more resources because it could a leader?

**Reply by chen-keinan** [reply]
I think so , need to be checked.
also easy to test , scale down to 1 and check it resources
